ui/src/app/components/rename-modal.tsx

```diff
@@ -24,9 +25,45 @@ const styles = reactStyles({
   }
 });
 
+/* Name validation for resources.
+ *
+ * Notebooks have characters that are disallowed. Specifically, Jupyter only disallows :/\
+ * but Terra disallows a larger list of characters.  Those characters are listed in
+ * `const notebookNameValidator` in this file:
+ * https://github.com/DataBiosphere/terra-ui/blob/dev/src/components/notebook-utils.js#L42
+ *
+ * Other AoU resource types do not have the same name restriction and only block slashes.
+ */
+export const nameValidationFormat = (existingNames, type) =>
+  type === ResourceType.NOTEBOOK ?
+    ({
+      presence: {allowEmpty: false},
+      format: {
+        pattern: /^[^#[\]*?:;@$%+=\\,/]*$/,
+        message: 'can\'t contain these characters: \r \r  @ # $ % * + = ? , [ ] : ; / \\ '
```

Comment:
Did you mean `\r \n`?

Reply:
Also, nit: could you please put these in the same order as is in the regex if possible?

Reply:
Good question. I copied that over from Saturn code but I'll remove it.